samples/Basic_Neopixel/app/application.cpp

```diff
@@ -188,7 +190,9 @@ void init()
 	WifiStation.config(WIFI_SSID, WIFI_PWD);
 	WifiStation.enable(true);
 	WifiAccessPoint.enable(false);
-	WifiStation.waitConnection(connect_Ok, 20, connect_Fail);
+	WifiEvents.onStationConnect(connect_Ok);
```

Comment:
Talking about 'focus': Please, remove the WifiEvents.onStationConnect callback and move its code to the callback for `WifiEvents.onStationGotIP(got_IP)` .  Example usage of `WifiEvents.onStationConnect` can be added to Basic_Wifi sample, which is the one dedicated to Wifi connectivity and Wifi Events handling.